homeassistant/components/dlna_dmr/config_flow.py

```diff
@@ -40,6 +40,19 @@
 FlowInput = Optional[Mapping[str, Any]]
 
 
+def _parse_hostname(ssdp_location: str | None, default: str | None = None) -> str:
+    if ssdp_location is None:
+        if default is None:
+            raise ValueError("ssdp_location is None")
+        return default
+    url = urlparse(ssdp_location)
+    if url.hostname is None:
+        if default is None:
+            raise ValueError("hostname is None")
+        return default
+    return url.hostname
```

Comment:
Obviously it complains about code coverage on this...
But if it agreed upon, I can move to a core helper class and run some tests.

Reply:
I have reverted the changes to `_parse_hostname` to put them in a separate PR: #60839